jupyter_server_ydoc/app.py

```diff
@@ -41,14 +41,47 @@ class YDocExtension(ExtensionApp):
     )
 
     ystore_class = Type(
-        default_value=JupyterSQLiteYStore,
+        default_value=sqlite_ystore_factory(),
         klass=BaseYStore,
         config=True,
         help="""The YStore class to use for storing Y updates. Defaults to JupyterSQLiteYStore,
         which stores Y updates in a '.jupyter_ystore.db' SQLite database in the current
         directory, and clears history every 24 hours.""",
     )
 
+    sqlite_ystore_db_path = Unicode(
+        ".jupyter_ystore.db",
+        config=True,
+        help="""The path to the YStore database. Defaults to '.jupyter_ystore.db' in the current
+        directory. Only applicable if the YStore is an SQLiteYStore.""",
+    )
+
+    @observe("sqlite_ystore_db_path")
+    def _observe_sqlite_ystore_db_path(self, change):
+        if issubclass(self.ystore_class, SQLiteYStore):
+            self.ystore_class.db_path = change["new"]
+        else:
+            raise RuntimeError(
+                f"ystore_class must be an SQLiteYStore to be able to set sqlite_ystore_db_path, not {self.ystore_class}"
+            )
+
+    sqlite_ystore_document_ttl = Int(
```

Comment:
Wait, so why not just:

1. Have `JupyterSQLiteYStore` be a subclass of both `Configurable` and `SQLiteYStore`, defined at file scope rather than in a factory
2. Define these traits on that class itself
3. Revert back to `default_value=JupyterSQLiteYStore` on the `ystore_class` trait

I don't think adding configurable traits on the extension app class is the right approach here, given that they're specific to whether the user is using a subclass of `SQLiteYStore`.